crates/types/protobuf/restate/cluster.proto

```diff
@@ -92,8 +92,15 @@ message BifrostProvider {
   uint32 target_nodeset_size = 3;
 }
 
+message PartitionReplication {
+  oneof replication {
+    ReplicationProperty limited = 1;
+    google.protobuf.Empty everywhere = 2;
+  }
```

Comment:
It's not super critical because we don't use `ClusterConfiguration` on the hot path, but I wanted to share that `oneofs` usually lead to deeper nested structures which can impact serde performance negatively.

Reply:
I was trying to avoid adding an extra fields directly on the ClusterConfiguration since it will feel out of place and allows both fields (partition_replication and everywhere) to be set at the same time.

But it seems i used oneof in the wrong way here for compatibility. I should have instead replaced the partition_replication in-place instead of creating a new message that encapsulate the oneof! sorry about that.

So maybe I will just add an enum instead

Reply:
YMMV, but oneof is not different from encoding multiple attributes as messages. I wouldn't be worried about it as long as the nesting is not incredibly deep.